qcodes/instrument_drivers/stanford_research/SR830.py

```diff
@@ -656,6 +747,10 @@ def _set_ch_display(self, channel: int, disp: str) -> None:
         ratio_val = int(self.ask(f'DDEF ? {channel}').split(',')[1])
         self.write(f'DDEF {channel}, {disp_int}, {ratio_val}')
         self._buffer_ready = False
+        # we update the unit of the datatrace
+        # according to the choice of channel
+        params = self.parameters
+        params[f'ch{channel}_datatrace'].update_unit()
```

Comment:
```suggestion
        dataparam = self.parameters[f'ch{channel}_datatrace']
        assert isinstance(dataparam,  GeneratedSetPoints)
        dataparam.update_unit()
```

Mypy cannot see that this parameter is a GeneratedSetPoints parameter so we can help it by asserting that as shown here

Reply:
thanks it worked (replacing GeneratedSetPoints by ChannelTrace)
I was confused because I had tryed:
        params = self.parameters
        assert isinstance(params[f'ch{channel}_datatrace'], ChannelTrace)
        params[f'ch{channel}_datatrace'].update_unit()
which did not work.

Now I am left with mypy errors for the init function of GeneratedSetPoint and ChannelTrace

Reply:
Yes you have to make that a variable before using the assert. Mypy will not look through the indirection of the dict lookup